### PR TITLE
feat: connect confirmation

### PR DESCRIPTION
**Pull request by KovaCro**

### CODE REVIEW

File: packages/lib/src/confirm/render-confirm.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    confirmEl.addEventListener(event, () => clear(event));
+  });
+
+  if (options.closable) {
+    confirmEl.addEventListener('close', () => clear('close'));
+  }
+
+  if (options.duration) {
+    setTimeout(() => clear('timeout'), options.duration);
+  }
```
Comment (flauc) [inline packages/lib/src/confirm/render-confirm.ts:39]: No timeout is necessary for the confirmation. It should persist on screen until actioned on.

File: packages/lib/src/confirm/render-confirm.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import type { ConfirmOptions } from '../types/confirm-options.interface';
+
+export async function renderConfirm(
+  options?: ConfirmOptions,
+  callback?: (action?: string) => void
+) {
+  options = {
+    host: document.body,
+    duration: null,
+    closable: true,
```
Comment (flauc) [inline packages/lib/src/confirm/render-confirm.ts:10]: Let's set `closable` to `false` by default.

File: packages/lib/src/confirm/confirm.wc.svelte [hunk 3]
```diff
@@ -1,57 +1,43 @@
...
-        left: 0;
-        display: flex;
-        justify-content: center;
-        align-items: center;
-        width: 100vw;
-        height: 100vh;
-        background-color: rgba(0,0,0,.2);
-    }
+  export let title = '';
+  export let message = '';
```
Comment (flauc) [inline packages/lib/src/confirm/confirm.wc.svelte:13]: Let's make the `Yes` and `No` wording customizable as well.

File: packages/lib/src/confirm/confirm.wc.svelte [hunk 4]
```diff
@@ -1,57 +1,57 @@
...
 
-    h1 {
-        font-size: 1.5rem;
-        font-weight: bold;
-    }
-</style>
\ No newline at end of file
+<svelte:window on:keydown|preventDefault={(e) => {
+    if(e.key == 'Escape') dispatch('close')
+}} />
```
Comment (flauc) [inline packages/lib/src/confirm/confirm.wc.svelte:23]: Don't we need to check if `closable` is `true`?

**Review by flauc — CHANGES_REQUESTED** [review]

File: packages/lib/src/confirm/confirm.wc.svelte (hunk 4 above)
Comment (KovaCro) [reply packages/lib/src/confirm/confirm.wc.svelte:23]: If `closable` is `false`, the event listener on `'close'` does not get attached.

Is it a good practice or should I change the approach?

Comment (flauc) [reply packages/lib/src/confirm/confirm.wc.svelte:23]: That's great and it should stay but the problem is the component should be usable without the utility function, If someone want's to integrate it with their system directly.

File: packages/lib/src/confirm/confirm.wc.svelte [hunk 5]
```diff
@@ -29,7 +31,7 @@
   <div
     class="bg-slate-50 px-12 py-16 rounded-xl max-w-sm"
     use:clickOutside
-    on:click_outside={() => dispatch('close')}
+    on:click_outside={() => {if(closable) dispatch('close')}}
```
Comment (flauc) [inline packages/lib/src/confirm/confirm.wc.svelte:34]: You can simplify this to just 

```
on:click_outside={() =>closable && dispatch('close')}
```

File: packages/lib/src/confirm/confirm.wc.svelte [hunk 6]
```diff
@@ -15,11 +15,13 @@
...
   export let accept = '';
 
+  export let closable = false;
+
   const dispatch = createEventDispatcher();
 </script>
 
 <svelte:window on:keydown|preventDefault={(e) => {
-    if(e.key == 'Escape') dispatch('close')
+    if(e.key == 'Escape' && closable) dispatch('close')
```
Comment (flauc) [inline packages/lib/src/confirm/confirm.wc.svelte:24]: Just to make it a bit cleaner: 

```
<svelte:window on:keydown|preventDefault={(e) => e.key === 'Escape' && closable && dispatch('close')} />
 ```

**Review by flauc — CHANGES_REQUESTED** [review]

**Pull request merged**